### PR TITLE
Remove shh and shhext from APIModules

### PR DESCRIPTION
**Pull request by sebastiandelgado**
Addresses #912. By removing these two modules from APIModules, we disallow dapps from using dangerous methods like shh.SetMinPoW

### CODE REVIEW

**Comment by sebastiandelgado** [comment]
Still can't tell if I'm breaking any tests for sure with this change. Will update this once #992 gets merged and fixes the test issues on the develop branch.

File: geth/params/defaults.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ const (
 
 	// APIModules is a list of modules to expose via any type of RPC (HTTP, IPC, in-proc)
```
Comment (pedropombeiro) [inline geth/params/defaults.go]: Isn't this comment changing too, per @adambabik's previous review?

**Comment by adambabik** [comment]
```
--- FAIL: TestRPCTestSuite (7.58s)
    --- FAIL: TestRPCTestSuite/TestCallRawResult (2.57s)
	Error Trace:	rpc_test.go:137
	Error:      	Not equal: 
	            	expected: "{\"jsonrpc\":\"2.0\",\"id\":67,\"result\":\"6.0\"}"
	            	actual: "{\"jsonrpc\":\"2.0\",\"id\":67,\"error\":{\"code\":-32601,\"message\":\"The method shh_version does not exist/is not available\"}}"
```

This is a valid error. The method needs to be changed or `client := s.StatusNode.RPCPrivateClient()` should be used.

**Comment by adambabik** [comment]
There are more tests related to the RPC client:
```
--- FAIL: TestMailServiceSuite (4.47s)
    --- FAIL: TestMailServiceSuite/TestShhextRequestMessagesRPCMethodAvailability (4.45s)
	Error Trace:	mailservice_test.go:43
	Error:      	Error message not equal:
	            	expected: "invalid mailServerPeer value: invalid URL scheme, want \"enode\""
	            	actual: "The method shhext_requestMessages does not exist/is not available"
```

Please run them first locally. This time `go test -timeout 20m ./t/e2e/whisper/... -network=StatusChain -race` fails.

**Comment by sebastiandelgado** [comment]
@adambabik thanks. Theres a bunch more failed tests in `./t/e2e/whisper/...` too. Was just having a hard time finding what tests were failing on develop and which ones were failing on my branch specifically. Updated the PR to WIP while I address all of these this afternoon

**Comment by sebastiandelgado** [comment]
This PR is still a WIP

Updated a bunch of tests to use `RPCPrivateClient` instead of `RPCClient`

TODO:
 -  Figure out if all the tests are still valid when using `RPCPrivateClient` instead of `RPCClient`. Pretty sure several of them aren't valid tests until I update the source code to use `RPCPrivateClient` as well
 -  I still have the below whisper e2e test failing:

```
--- FAIL: TestMailServiceSuite (2.55s)
    --- FAIL: TestMailServiceSuite/TestShhextRequestMessagesRPCMethodAvailability (2.55s)
        Error Trace:    mailservice_test.go:63
    	Error:      	"{"jsonrpc":"2.0","id":1,"error":{"code":-32601,"message":"The method shhext_requestMessages does not exist/is not available"}}
    	            	" does not contain "invalid mailServerPeer value"
```

**Comment by sebastiandelgado** [comment]
All tests are passing now. Would be good to get some context on which tests shouldn't be allowed to use RPCPrivateClient and which should be ok to add "shh" or "shhext" manually in the node config in the tests, similar to mailservice_test.go

File: t/e2e/rpc/rpc_test.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func (s *RPCTestSuite) TestCallRPC() {
 
 		s.NoError(s.StatusNode.Start(nodeConfig))
 
-		rpcClient := s.StatusNode.RPCClient()
+		rpcClient := s.StatusNode.RPCPrivateClient()
```
Comment (adambabik) [inline t/e2e/rpc/rpc_test.go]: As this test is called `TestCallRPC`, we can change its name to `TestCallPrivateRPC` and create another one for testing `CallRPC` (the latter thing as a separate issue probably).

File: t/e2e/whisper/mailservice_test.go [hunk 3]
```diff
@@ -31,10 +31,11 @@ func (s *MailServiceSuite) TestShhextRequestMessagesRPCMethodAvailability() {
 
 	s.StartTestNode(func(config *params.NodeConfig) {
 		config.RPCEnabled = true
+		config.APIModules = "shhext"
 	})
 	defer s.StopTestNode()
 
-	client := s.StatusNode.RPCClient()
+	client := s.StatusNode.RPCPrivateClient()
```
Comment (adambabik) [inline t/e2e/whisper/mailservice_test.go]: If `shhext` is set in `APIModules` list, then it should work fine here with `RPCClient`, right?

**Review by adambabik — APPROVED** [review]
LGTM

Check out two suggestions.

**Pull request merged**